client/src/main/java/io/pravega/client/segment/impl/SegmentMetadataClient.java

```diff
@@ -19,18 +19,16 @@
     /**
      * Returns info for the current segment.
      *
-     * @param delegationToken token to be passed on to segmentstore for validation.
```

Comment:
These changes take delegationToken via the constructor of `SegmentMetadataClient` rather than as parameter to individual APIs. This means that the life time of both of these should be same. But it is possible that user can hold the `SegmentMetadataClient` object for a long time. (e.g. in RevisionedStreamClientImpl the meta object may live longer than the token). It makes sense to send delegationToken through each call and refresh it if required.

Reply:
After 1:1 discussions, I found that the behavior after expiry is same as other usages of delegation token. This change is not a blocker for the PR.